src/recipes/editor-plugin/inspector-plugins.md

```diff
@@ -0,0 +1,342 @@
+        let num = rand::thread_rng().gen_range(0..100);
+        godot_print!("Randomize! {} for {}", num, property_name);
+        self.base_mut()
+            .emit_changed(property_name, num.to_variant().to());
+        if let Some(mut button) = self.button.clone() {
+            let text = format!("Randomize: {}", num);
+            button.set_text(text.into());
+        } else {
+            // Print error of something went wrong
+            godot_error!("Button wasn't found in handle_press");
+        }
+    }
+}
+```
+
+
+## Add Inspector plugin
+
+Now we need to connect this editor to fields with an integer type.
+To do this, we need to create an EditorInspectorPlugin.
+
+```rust
+#[derive(GodotClass)]
+#[class(tool, init, base=EditorInspectorPlugin)]
+struct RandomInspectorPlugin {
+    base: Base<EditorInspectorPlugin>,
+}
+```
+
+IEditorInspectorPlugin implementation:
+
+```rust
+#[godot_api]
+impl IEditorInspectorPlugin for RandomInspectorPlugin {
+      fn parse_property(
+        &mut self,
+        _object: Gd<Object>, // object that inspecting
+        value_type: VariantType,
+        name: GString,
+        _hint_type: global::PropertyHint,
+        _hit_name: GString,
+        _flags: global::PropertyUsageFlags,
+        _wide: bool,
+    ) -> bool {
+        if value_type == VariantType::INT {
+            self.base_mut()
+                .add_property_editor(name, RandomIntEditor::new_alloc().to_variant().to());
+            return true;
+        }
+
+        false
+    }
+
+    fn can_handle(&self, _object: Gd<Object>) -> bool {
+        true
+    }
+}
+```
+
+If `parse_property` returns `true`, the editor will be created and replace the current
+representation; if not, it's necessary to return `false`.
+This allows for specific control over where and how processing occurs.
+
+
+## Add Editor Plugin
+
+Only one thing left to do: define the editor plugin that will kick off all this magic!
+This can be a generic `EditorPlugin` or a more specific `InspectorEditorPlugin`, depending
+on what you want to achieve.
+
+
+```rust
+#[derive(GodotClass)]
+#[class(tool, init, editor_plugin, base=EditorPlugin)]
+struct RustEditorPlugin {
+    base: Base<EditorPlugin>,
+    random_inspector: Gd<RandomInspectorPlugin>,
+}
+```
+
+```rust
+#[godot_api]
+impl IEditorPlugin for RustEditorPlugin {
+    fn enter_tree(&mut self) {
+        // create our inspector plugin and save it to next remove
+        self.random_inspector = RandomInspectorPlugin::new_gd();
+        self.to_gd()
+            .add_inspector_plugin(self.random_inspector.to_variant().to());
+    }
+
+    fn exit_tree(&mut self) {
+        self.to_gd()
+            .remove_inspector_plugin(self.random_inspector.to_variant().to());
+    }
+}
+```
+
+
+```admonish danger title="Troubleshooting"
+Sometimes after compilation, you may encounter errors or panic. Most likely, all you need to do is simply **restart** the Godot Editor.
+```
```

Comment:
Could be admonish `note`. Warnings/dangers are not to explain what errors mean to the user, but to make them aware of potential pitfalls.